### PR TITLE
improve assertions #1017

### PR DESCRIPTION
**Pull request by yoshikawaa**
Please review #1017.

#### Changes Policy

* Corrected assertions what should be done in `Matcher` instead of `actual` (verification target).  
  However, the part that is intended to execute the  method itself of the test target class (ex. `equals`, `toString`) is not modified.
* Improved to use `assertThrows` where try-catch is used for exception assertion.
* Redundant assertions and tests combined to make it compact.

### CODE REVIEW

File: terasoluna-gfw-common-libraries/terasoluna-gfw-codepoints/src/test/java/org/terasoluna/gfw/common/codepoints/validator/ConsistOfValidatorTest.java [hunk 1]
```diff
@@ -37,379 +36,313 @@
...
         Set<ConstraintViolation<Name_Simple>> violations = validator.validate(
                 name);
 
-        assertThat(violations, is(notNullValue()));
-        assertThat(violations.size(), is(1));
-
-        ConstraintViolation<Name_Simple> v = violations.iterator().next();
-        assertThat(v.getPropertyPath().toString(), is("lastName"));
-        assertThat(v.getMessage(), is("not consist of specified code points"));
+        assertThat(violations, containsInAnyOrder( //]
```
Comment (btkatoufm) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-codepoints/src/test/java/org/terasoluna/gfw/common/codepoints/validator/ConsistOfValidatorTest.java:87]: There is an unnecessary comment "]".

File: terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/test/java/org/terasoluna/gfw/common/validator/constraints/CompareTest.java [hunk 2]
```diff
@@ -523,72 +501,67 @@ public void testSpecifyNotComparableLeft() throws Throwable {
...
+                    new Pair(100, 101)));
+
+            violations = validator.validate(form);
+            assertThat(violations, containsInAnyOrder( //
+                    allOf( //
+                            hasProperty("propertyPath", hasToString(
+                                    "listProperty[0].<list element>.left")), //
+                            hasProperty("message", is(String.format(
+                                    MESSAGE_VALIDATION_ERROR, "left",
+                                    "right")))), allOf( //
```
Comment (btkatoufm) [inline terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/test/java/org/terasoluna/gfw/common/validator/constraints/CompareTest.java:556]: The format is broken.
It is recommended to put `//` between `),` and `allOf(` to ensure formatting.

File: terasoluna-gfw-common-libraries/terasoluna-gfw-codepoints/src/test/java/org/terasoluna/gfw/common/codepoints/validator/ConsistOfValidatorTest.java (hunk 1 above)
Comment (yoshikawaa) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-codepoints/src/test/java/org/terasoluna/gfw/common/codepoints/validator/ConsistOfValidatorTest.java]: Thx! I'll fix it. 3feceda

File: terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/test/java/org/terasoluna/gfw/common/validator/constraints/CompareTest.java (hunk 2 above)
Comment (yoshikawaa) [reply terasoluna-gfw-common-libraries/terasoluna-gfw-validator/src/test/java/org/terasoluna/gfw/common/validator/constraints/CompareTest.java]: Thx! I fixed all with grep `allOf(`. 3feceda

**Pull request merged**